extensions/amp-addthis/0.1/post-message-dispatcher.js

```diff
@@ -57,12 +57,13 @@ export class PostMessageDispatcher {
    * Utility method to parse out the data from the supplied `postMessage` event.
    * @param {!Event} event
    * @private
+   * @return {?JsonObject|undefined}
    */
   getMessageData_(event) {
     const data = getData(event);
 
     if (isObject(data)) {
-      return data;
+      return /** @type {!JsonObject} */ (data);
```

Comment:
Can you annotate the return type of getData instead?

Reply:
The return type of `getData` is already specified. It inherits the return type of `{?JsonObject|string|undefined}`.

The cast is valid because `isObject(data)` is `true`, and is required because of this inconsistent return type error:

```js
Type checking failed:
extensions/amp-addthis/0.1/post-message-dispatcher.js:66: ERROR - [JSC_TYPE_MISMATCH] inconsistent return type
found   : (JsonObject|null|string|undefined)
required: (JsonObject|null|undefined)
      return data;
             ^^^^
```

Reply:
Is there some sort of `isJsonObject` you could use instead of `isObject` so the compiler can be checking this for us?

Reply:
There isn't.

Reply:
I've opened #23567 for this.